File: src/main/java/lotto/domain/PurchaseAmount.java
package lotto.domain;

import static lotto.utils.ErrorMessage.PRICE_LACK_ERROR;
import static lotto.utils.ErrorMessage.PRICE_UNIT_ERROR;

public class PurchaseAmount {

    public static final int LOTTO_PRICE = 1000;

    private final long userPrice;

    private PurchaseAmount(long userPrice) {
        validateMinimumPrice(userPrice);
        validatePriceUnit(userPrice);
        this.userPrice = userPrice;
    }

    public static PurchaseAmount from(long userPrice) {
        return new PurchaseAmount(userPrice);
    }

    private void validateMinimumPrice(long userPrice) {
        if (userPrice < LOTTO_PRICE) {
            throw new IllegalArgumentException(PRICE_LACK_ERROR.getMessage());
        }
    }

    private void validatePriceUnit(long userPrice) {
        if (userPrice % LOTTO_PRICE != 0) {
            throw new IllegalArgumentException(PRICE_UNIT_ERROR.getMessage());
        }
    }

    public long getLottoCount() {
        return userPrice / LOTTO_PRICE;
    }

}

File: src/main/kotlin/lotto/domain/PurchaseAmount.kt
package lotto.domain

import lotto.utils.ErrorMessage.*

data class PurchaseAmount private constructor(
    val userPrice: Long
) {

    init {
        validateMinimumPrice(userPrice)
        validatePriceUnit(userPrice)
    }

    companion object {
        const val LOTTO_PRICE: Int = 1000

        fun from(userPrice: Long): PurchaseAmount = PurchaseAmount(userPrice)

        private fun validateMinimumPrice(userPrice: Long) {
            if (userPrice < LOTTO_PRICE) {
                throw IllegalArgumentException(PRICE_LACK_ERROR.getMessage())
            }
        }

        private fun validatePriceUnit(userPrice: Long) {
            if (userPrice % LOTTO_PRICE != 0L) {
                throw IllegalArgumentException(PRICE_UNIT_ERROR.getMessage())
            }
        }
    }

    fun calculateLottoCount(): Long = userPrice / LOTTO_PRICE
}